parlai/agents/example_seq2seq/example_seq2seq.py

```diff
@@ -262,56 +242,36 @@ def predict(self, xs, ys=None, is_training=False):
 
         return predictions
 
-    def vectorize(self, observations):
-        """Convert a list of observations into input & target tensors."""
-        is_training = any(('labels' in obs for obs in observations))
-        # utility function for padding text and returning lists of indices
-        # parsed using the provided dictionary
-        xs, ys, labels, valid_inds, _, _ = PaddingUtils.pad_text(
-            observations, self.dict, end_idx=self.END_IDX,
-            null_idx=self.NULL_IDX, dq=False, eval_labels=True)
-        if xs is None:
-            return None, None, None, None, None
-
-        # move lists of indices returned above into tensors
-        xs = torch.LongTensor(xs)
-        if self.use_cuda:
-            xs = xs.cuda()
-        xs = Variable(xs)
-
-        if ys is not None:
-            ys = torch.LongTensor(ys)
-            if self.use_cuda:
-                ys = ys.cuda()
-            ys = Variable(ys)
-
-        return xs, ys, labels, valid_inds, is_training
-
     def batch_act(self, observations):
-        batchsize = len(observations)
+        batch_size = len(observations)
         # initialize a table of replies with this agent's id
-        batch_reply = [{'id': self.getID()} for _ in range(batchsize)]
+        batch_reply = [{'id': self.getID()} for _ in range(batch_size)]
+
+        is_training = any(['labels' in obs for obs in observations])
+
+        vec_obs = [self.vectorize(obs)
+                   for obs in observations]
 
-        # convert the observations into batches of inputs and targets
-        # `labels` stores the true labels returned in the `ys` vector
-        # `valid_inds` tells us the indices of all valid examples
-        # e.g. for input [{}, {'text': 'hello'}, {}, {}], valid_inds is [1]
-        # since the other three elements had no 'text' field
-        xs, ys, labels, valid_inds, is_training = self.vectorize(observations)
+        xs, ys, labels, valid_inds = self.map_valid(vec_obs)
 
         if xs is None:
-            # no valid examples, just return empty responses
             return batch_reply
 
         predictions = self.predict(xs, ys, is_training)
 
-        # maps returns predictions back to the right `valid_inds`
-        # in the example above, a prediction `world` should reply to `hello`
-        PaddingUtils.map_predictions(
-            predictions.cpu().data, valid_inds, batch_reply, observations,
-            self.dict, self.END_IDX, labels=labels,
-            answers=labels, ys=ys.data if ys is not None else None,
-            report_freq=self.opt.get('report_freq', 0))
+        unpermute_pred = self.unmap_valid(predictions.cpu().data,
```

Comment:
"permute"

`predictions.cpu().data` => `predictions`, and then can probably move this call onto one line